Add 10.11 Studio Pro release notes (#7771)
mendix/docs: content/en/docs/releasenotes/studio-pro/10/10.11.md

* We added a preview of the auto-generated API documentation for published OData services to Studio Pro. This allows you to see the resulting REST endpoints of the OData service while you are modeling it.

### Java 21

We now support Java 21 in Studio Pro:
* This release ships with Java Development Kit (JDK) 21.
* You can now configure your application to use Java 21 in the **App Settings**.
* In **Preferences** you can configure a different JDK for each supported version of Java.
* Targeting Java 21 allows the project to use language constructs and APIs that were introduced with Java 21.
* Newly created projects will be configured to use Java 21 by default.

We recommend that you upgrade apps and Marketplace modules to Java 21.

### Improvements

* We changed the log level of the *Tasks cannot be executed anymore due to model changes* message in the task queue from **critical** to **error**.  (Ticket 199565)
* We ensure that the `vendorlib-sbom.json` file does not change when dependencies do not change, reducing the number of git conflicts related to this file. (Ticket 208993)
* We now pass a render mode to the pluggable widgets API to indicate which mode the page editor is in.

> * We added a `CanApplyJumpTo` attribute to the `System.Workflow` entity that indicates whether it is allowed to apply a jump-to option to the workflow in its current state.

Reviewer: "whether it is allowed" - what is it? Does it mean:
```suggestion
* We added a `CanApplyJumpTo` attribute to the `System.Workflow` entity to indicate whether you can apply a jump-to option to the workflow in its current state.
```

* We made some performance improvements when finding usages in protected modules.

> * We added a **Support Tools** submenu to the **Help** menu, this submenu contains the following tools: Performance logging, Profiling, and Optimize MPR.

Reviewer: Documentation?

* We improved parameter validation while trying REST requests in consumed REST service editor (Beta). Requests having parameter values with spaces in URL will now raise errors as they do when used in microflow action.

> * We upgraded react-navigation to version 6.x.x and extended navigation transitions options for Studio Pro.

Reviewer: Should this not have a real version? **6.x.x** looks like a placeholder - I would think there is an actual version which is shipped?

* We now support keyboard hotkeys to copy, cut, paste, undo, redo, and delete in MendixChat (Windows only).
* We made sure that the right-click context menu with the **Cut**, **Copy**, **Paste**, and **Delete** options is available on the MendixChat feedback page.
* We added rendering support for Markdown tables, code blocks, and headers in MendixChat.
* We added a small delay before showing the MxAssist Logic Bot blue dot.
* We fixed memory leaks that happened when closing and opening applications in Studio Pro.
* We do not allow you to provide an invalid or duplicate request name in the consumed REST service document.
* We made some performance improvements to several operations in Studio Pro, including renaming entities, pasting in the domain model editor, copying and duplicating documents, and importing packages.
* We improved relay-outing in the logic editors: when adding objects to a loop, it only expands in the relevant direction(s).
* We have made a couple of small performance improvements to the Logic Bot.
* The menu items in the **View** menu are now ordered alphabetically and have individually assigned keyboard shortcuts.

> * We improved consistency checks for parameters in the Consumed REST Request (beta) microflow action. Parameters with different casing are also marked as duplicate now.
> * We added support for parameters in Path URL inside a Consumed REST service (beta) document.

Reviewer: Are some of these UX elements?
```suggestion
* We improved consistency checks for parameters in the **Consumed REST Request** (beta) microflow action. Parameters with different casing are also marked as duplicate now.
* We added support for parameters in **Path URL** inside a **Consumed REST service** (beta) document.
```
There is also a capitalization difference between "REST Request" and "REST service"

* We fixed an issue where, when migrating a column from Float to Decimal, the default value was converted to Float before being inserted, which caused improper rounding.
* We fixed an issue in OQL where subtracting the `null` literal caused subsequent `null` literal usages to throw a parsing exception.
* When using MySQL as the database, setting a query timeout works again.
* We resolved an issue in Studio Pro where an error occurred when attempting to open **Preferences** while creating a new app.
* We fixed an issue where aborting a workflow resulted in an error if the workflow had an ongoing activity that had just finished.
* We fixed an issue where a pluggable widget with a [selection property](/apidocs-mxsdk/apidocs/pluggable-widgets-property-types/#selection) linked to a [data source property](/apidocs-mxsdk/apidocs/pluggable-widgets-property-types/#datasource) which was not required resulted in an exception when the data source was not configured. 
* We fixed an exception that happened upon closing a non-version-controlled app (after you previously opened a version controlled app and did not restart Studio Pro).

> * We fixed an issue in the [Core.RetrieveByPath](https://apidocs.rnd.mendix.com/10/runtime/com/mendix/core/Core.html) Runtime API, where having a new external object and retrieving over an association from the child's side resulted in an error.

Reviewer: I am never sure what we mean by child. Do we mean the side which doesn't own the association?

Perhaps it is just me?
